CMakeLists.txt

```diff
@@ -0,0 +1,237 @@
+        BYPRODUCTS ${GENERATED_SOURCE_FILES}
+)
+
+include(SourceList REQUIRED)
+include(Endian REQUIRED)
+add_endian_definition()
+
+# Needed by rev.cpp
+include_directories(${CMAKE_SOURCE_DIR}/src)
+# Needed by everything that uses Squirrel
+include_directories(${CMAKE_SOURCE_DIR}/src/3rdparty/squirrel/include)
+
+include(CompileFlags REQUIRED)
+compile_flags()
+
+add_executable(openttd WIN32 ${GENERATED_SOURCE_FILES})
+# All other files are added via target_sources()
+
+include(AddCustomXXXTimestamp REQUIRED)
+add_subdirectory(${CMAKE_SOURCE_DIR}/src)
+add_subdirectory(${CMAKE_SOURCE_DIR}/media/baseset)
+
+add_dependencies(openttd
+    find_version)
+
+target_link_libraries(openttd
+    openttd::languages
+    openttd::settings
+    openttd::basesets
+    Threads::Threads
+)
+
+if (IPO_FOUND)
+    set_target_properties(openttd PROPERTIES INTERPROCEDURAL_OPTIMIZATION_RELEASE True)
+    set_target_properties(openttd PROPERTIES INTERPROCEDURAL_OPTIMIZATION_MINSIZEREL True)
+    set_target_properties(openttd PROPERTIES INTERPROCEDURAL_OPTIMIZATION_RELWITHDEBINFO True)
+endif (IPO_FOUND)
+set_target_properties(openttd PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/bin")
+process_compile_flags()
+
+if (APPLE OR UNIX)
+    add_definitions(-DUNIX)
+endif (APPLE OR UNIX)
+
+link_library(PNG ENCOURAGED)
+link_library(ZLIB ENCOURAGED)
+link_library(LIBLZMA ENCOURAGED)
+link_library(LZO ENCOURAGED)
+link_library(XDG_BASEDIR)
+
+if (NOT OPTION_DEDICATED)
+    link_library(FLUIDSYNTH)
+    link_library(SDL)
+    link_library(ALLEGRO)
+    link_library(FREETYPE)
+    link_library(FONTCONFIG)
+    link_library(ICU_lx)
+    link_library(ICU_i18n)
+endif (NOT OPTION_DEDICATED)
+
+if (APPLE)
+    link_library(Iconv)
+
+    target_link_libraries(openttd ${AUDIOTOOLBOX_LIBRARY})
+    target_link_libraries(openttd ${AUDIOUNIT_LIBRARY})
+    target_link_libraries(openttd ${CARBON_LIBRARY})
+    target_link_libraries(openttd ${COCOA_LIBRARY})
+
+    add_definitions(-DWITH_COCOA)
+    add_definitions(-DENABLE_COCOA_QUARTZ)
+    add_definitions(-DNO_QUICKTIME)
+endif (APPLE)
+
+if (NOT PERSONAL_DIR STREQUAL "(not set)")
+    add_definitions(-DWITH_PERSONAL_DIR)
+    add_definitions(-DPERSONAL_DIR="${PERSONAL_DIR}")
+endif (NOT PERSONAL_DIR STREQUAL "(not set)")
+
+if (NOT SHARED_DIR STREQUAL "(not set)")
+    add_definitions(-DWITH_SHARED_DIR)
+    add_definitions(-DSHARED_DIR="${SHARED_DIR}")
+endif (NOT SHARED_DIR STREQUAL "(not set)")
+
+if (NOT GLOBAL_DIR STREQUAL "(not set)")
+    add_definitions(-DGLOBAL_DATA_DIR="${GLOBAL_DIR}")
+endif (NOT GLOBAL_DIR STREQUAL "(not set)")
+
+if (SSE_FOUND)
+    add_definitions(-DWITH_SSE)
+endif (SSE_FOUND)
+
+add_definitions_based_on_options()
+
+if (WIN32)
+    add_definitions(-DUNICODE -D_UNICODE)
+    add_definitions(-DWITH_UNISCRIBE)
+
+    target_link_libraries(openttd
+        ws2_32
+        winmm
+        imm32
+    )
+endif (WIN32)
+
+if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+    add_definitions(-D_SQ64)
```

Comment:
I have not looked into the actual code, but can't you just check this there? Even if it is an ordinary
```c
if(sizeof(void*) == 8)
```
the compiler will optimize it away as it will detect that it will not change at runtime.

Reply:
Yes, we could; but Squirrel is a 3rdparty library, so these kind of changes can be hefty. It is simply easier to do it like this. But for sure this needs further attention towards the future :)